src/lib/strava/index.ts

```diff
@@ -0,0 +1,64 @@
+import {SECURE_STORE_CURRENT_USER_KEY} from '../../constants';
+import * as SecureStore from 'expo-secure-store';
+import {authorize, refreshToken, Athlete, Token} from './api';
+
+// Note loadToken can't be in the same
+// module as .refresh because then mocking doesn't
+// work so we've split it into ./api + index
+async function loadToken(): Promise<Token | null> {
+  const tokenData = await SecureStore.getItemAsync(
+    SECURE_STORE_CURRENT_USER_KEY,
+  );
+
+  if (tokenData) {
+    // ensure its fresh.
+    const token = JSON.parse(tokenData);
+
+    // strava access tokens has a 6hr ttl so we check if they
```

Comment:
This is interesting to know. We might need to run a login or refresh on wifi connect. Thinking about the case that someone rides for longer than 6 hours with no internet.